app/imports/components/posts/NewPost.jsx

```diff
@@ -106,25 +115,15 @@ export class NewPost extends Component {
       ctx.drawImage(this.phantomImage, 0, 0);
 
       // export base64
-      this.setState({
-        fixedOrientation: true,
-        photoRotatedSrc: canvas.toDataURL(),
-        finalWidth: canvas.width,
-        finalHeight: canvas.height,
-      });
-    } else {
-      this.setState({
-        fixedOrientation: true,
-        photoRotatedSrc: this.state.originalPhoto,
-        finalWidth: width,
-        finalHeight: height,
-      });
+      let mimeType = Meteor.settings.public.imageCompression.mimeType;
+      this.phantomImage.src = canvas.toDataURL(mimeType, 1);
     }
+
   }
 
   resizeImage() {
-    let imageWidth = this.state.finalWidth
-    let imageHeight = this.state.finalHeight
+    let imageWidth = this.phantomImage.width;
+    let imageHeight = this.phantomImage.height;
 
     // If image is larger than maxWidth
     if (imageWidth > Meteor.settings.public.imageCompression.maxWidth) {
```

Comment:
i think it's unlikely for a small image to be recompressed. examples of this case: screenshots, downloaded images; neither would need orientation fix or resizing and most probably they are  optimised jpgs, not as much but i think enough.